MobileCenterPush/MobileCenterPush/MSPush.m

```diff
@@ -258,6 +318,15 @@ - (BOOL)didReceiveRemoteNotification:(NSDictionary *)userInfo fromUserNotificati
   // The notification is not for Mobile Center if customData is nil. Ignore the notification.
   NSDictionary *customData = [userInfo objectForKey:kMSPushNotificationCustomDataKey];
   if (customData) {
+
+    // If Push is disabled, discard the notification.
+    if (![[self class] isEnabled]) {
+      MSLogVerbose(
+          [MSPush logTag],
+          @"Notification received while Push was enabled but Push is not disabled now, discard the notification.");
```

Comment:
`Notification received while Push was enabled but it is disabled now, discard the notification.`